geometry/pair.hpp

```diff
@@ -66,7 +66,7 @@ struct enable_if_vector;
 template<typename T1, typename T2, typename U>
 struct enable_if_vector<
     Pair<T1, T2>, U,
-    std::enable_if_t<std::conjunction_v<is_vector<T1>, is_vector<T2>>>>
+    std::enable_if_t<additive_group<T1> && additive_group<T2>>>
```

Comment:
I think that this should require vector spaces, otherwise the implementation of barycentre is problematic.

Reply:
See my other comment.